xarray/core/dataset.py

```diff
@@ -1324,7 +1332,7 @@ def copy(
     def _copy(
         self,
         deep: bool = False,
-        data: Mapping[Any, ArrayLike] | None = None,
```

Comment:
I assume the ArrayLike is false?

Reply:
yeah, the real condition is whatever Variable._copy allows.